app/com/lucidchart/open/cashy/controllers/BrowseController.scala

```diff
@@ -148,6 +156,66 @@ class BrowseController extends AppController {
   }
 
   /**
+   * Marks a hidden asset or folder as not hidden
+   */
+  def show(bucket: String, key: String) = AuthAction.authenticatedUser { implicit user =>
+    Action { implicit request =>
+      try {
+        if (key.endsWith("/")) {
+          // It is a folder
+          FolderModel.findByKey(bucket, key) match {
+            case Some(folder) => FolderModel.updateHidden(folder.id, false)
+            case None => FolderModel.createFolder(bucket, key, false)
```

Comment:
Because we weren't previously keeping track of folders because we didn't need to.  The only reason to keep track now is to see which ones are hidden.  So in the case here isn't a folder it isn't hidden and if we are hiding a folder that doesn't exist, we create it. 



> On May 5, 2015, at 10:52 AM, Mark Siebert <notifications@github.com> wrote:
> 
> In app/com/lucidchart/open/cashy/controllers/BrowseController.scala:
> 
> > @@ -148,6 +156,66 @@ class BrowseController extends AppController {
> >    }
> >  
> >    /**
> > +   * Marks a hidden asset or folder as not hidden
> > +   */
> > +  def show(bucket: String, key: String) = AuthAction.authenticatedUser { implicit user =>
> > +    Action { implicit request =>
> > +      try {
> > +        if (key.endsWith("/")) {
> > +          // It is a folder
> > +          FolderModel.findByKey(bucket, key) match {
> > +            case Some(folder) => FolderModel.updateHidden(folder.id, false)
> > +            case None => FolderModel.createFolder(bucket, key, false)
> Why is it that the None case for folders create a folder, but the None case for assets throws an exception?
> 
> —
> Reply to this email directly or view it on GitHub.
>